app/src/main/resources/application.properties

```diff
@@ -78,3 +78,41 @@ mp.openapi.servers=/api
 %prod.registry.events.kafka.config.enable.idempotence=true
 %prod.registry.events.kafka.config.retries=3
 %prod.registry.events.kafka.config.acks=all
+
+#Auth - disabled by default
+
+quarkus.oidc.enabled=true
+quarkus.oidc.tenant-enabled=${AUTH_ENABLED:false}
+
+registry.keycloak.url=${KEYCLOAK_URL:https://studio-auth.apicur.io/auth}
+registry.keycloak.realm=${KEYCLOAK_REALM:apicurio-local}
+
+quarkus.oidc.auth-server-url=${registry.keycloak.url}/realms/${registry.keycloak.realm}
+quarkus.oidc.client-id=${KEYCLOAK_API_CLIENT_ID:registry-api}
+
+registry.ui.config.auth.keycloak.url=${registry.keycloak.url}
+registry.ui.config.auth.keycloak.realm=${registry.keycloak.realm}
+registry.ui.config.auth.keycloak.clientId=${KEYCLOAK_UI_CLIENT_ID:apicurio-registry}
+registry.ui.config.auth.keycloak.onLoad=login-required
+
+quarkus.http.auth.policy.read-policy.roles-allowed=registry-user,registry-admin
+quarkus.http.auth.permission.user.enabled=${AUTH_ENABLED:false}
+quarkus.http.auth.permission.user.paths=/api/*,/public/*,/css/*,/js/*,/robots.txt
+quarkus.http.auth.permission.user.policy=read-policy
+quarkus.http.auth.permission.user.methods=GET,HEAD
+
+quarkus.http.auth.policy.modify-policy.roles-allowed=registry-admin
+quarkus.http.auth.permission.admin.enabled=${AUTH_ENABLED:false}
+quarkus.http.auth.permission.admin.paths=/api/*,/public/*,/css/*,/js/*,/robots.txt
+quarkus.http.auth.permission.admin.policy=modify-policy
+quarkus.http.auth.permission.admin.methods=PUT,POST,DELETE,PATCH
+
+quarkus.http.auth.permission.permit1.enabled=${AUTH_ENABLED:false}
+quarkus.http.auth.permission.permit1.paths=/health,/openapi,/metrics/*
+quarkus.http.auth.permission.permit1.policy=permit
+quarkus.http.auth.permission.permit1.methods=GET
+
+quarkus.http.auth.permission.authenticated.enabled=${AUTH_ENABLED:false}
+quarkus.http.auth.permission.authenticated.paths=/api/*
+quarkus.http.auth.permission.authenticated.policy=authenticated
```

Comment:
Can you iterate on this to match the description in the security ADR?  The roles defined there are a bit different than what we have here.  Also you'll need to make sure we're protecting both compatibility APIs as well (ccompat and ibmcompat).

Reply:
I have adapted the permissions and tested the compatibility APIs.

Reply:
I think there should be 3 roles, no? 

* Read-Only
* Modify allowed (standard access)
* Admin (same as above + can modify global rules)

I'm not seeing that here yet.

Reply:
I think that you have reviewed an outdated version of the file. I've added the three requires roles.

Reply:
You were right, sorry about that.  I was definitely looking at the old stuff.  My bad.